### PR TITLE
Make sure oc_xendit_order are filled

### PR DESCRIPTION
**Pull request by IreneGohtami**
OC 1.5:
- Invoice order are now inserted to oc_xendit_order
- CC orders are now inserted to oc_xendit_order
- Refactor `external_id` prefix = `opencart-xendit-`

OC 3.0:
- Invoice order are now inserted to oc_xendit_order
- CC orders are now inserted to oc_xendit_order
- Refactor `external_id` prefix = `opencart-xendit-`
- Font size reduced to 14px

* No more table oc_xendit_charge

### CODE REVIEW

File: opencart1.5.x/upload/catalog/model/payment/xendit.php [hunk 1]
```diff
@@ -8,18 +8,36 @@ public function getMethod($address, $total)
...
         return $this->db->getLastId();
     }
 
-    public function completeOrder($order_id)
+    public function completeOrder($order_id, $extra = '')
     {
-        $this->db->query("UPDATE `" . DB_PREFIX . "xendit_order` SET `status` = 'COMPLETED' WHERE `order_id` = '" . $order_id . "'");
-    }*/
+        $this->db->query("UPDATE `" . DB_PREFIX . "xendit_order` 
+                          SET `status` = 'COMPLETED', `xendit_paid_date` = NOW() $extra
```
Comment (candrasaputra) [inline opencart1.5.x/upload/catalog/model/payment/xendit.php]: PAID

File: opencart1.5.x/upload/catalog/model/payment/xendit.php [hunk 2]
```diff
@@ -8,18 +8,36 @@ public function getMethod($address, $total)
...
+                          WHERE `order_id` = '" . $order_id . "'");
+    }
+
+    public function cancelOrder($order_id)
+    {
+        $this->db->query("UPDATE `" . DB_PREFIX . "xendit_order` 
+                          SET `status` = 'FAILED', `xendit_cancelled_date` = NOW()
```
Comment (candrasaputra) [inline opencart1.5.x/upload/catalog/model/payment/xendit.php]: CANCELLED

**Comment by candrasaputra** [comment]
https://github.com/xendit/xendit-opencart-module/blob/d780f4adad2ce07821471240e5c18a611dc71fea/opencart3.0.x/upload/catalog/controller/extension/payment/xendit.php#L84

we need to handle this one, it will impacting our existing merchant

File: opencart1.5.x/upload/catalog/controller/payment/xendit.php [hunk 3]
```diff
@@ -79,10 +79,10 @@ public function process_notification() {
...
             $this->load->model('checkout/order');
 
-            $response = json_decode(file_get_contents('php://input'), true);
-            $invoice_id = $response['id'];
-            $external_id = $response['external_id'];
-            $order_id = str_replace(self::EXT_ID_PREFIX, "", $external_id);
+            $original_response = json_decode(file_get_contents('php://input'), true);
+            $invoice_id = $original_response['id'];
+            $external_id = $original_response['external_id'];
+            $order_id = preg_replace('/[^0-9]/', '', $external_id);
```
Comment (IreneGohtami) [inline opencart1.5.x/upload/catalog/controller/payment/xendit.php:85]: @candrasaputra for your previous concern, I have change it to extract numbers from string, so it doesn't depend on the prefix format anymore.

Comment (hakiemaul) [reply opencart1.5.x/upload/catalog/controller/payment/xendit.php:85]: whoa this is nice, we should use this in the future

**Review by hakiemaul — APPROVED** [review]

**Pull request merged**